frontend/src/container/GridGraphLayout/Graph/index.tsx

```diff
@@ -256,7 +257,29 @@ function GridCardGraph({
 					onView={handleOnView}
 					onDelete={handleOnDelete}
 				/>
-				<Spinner height="20vh" tip="Loading..." />
+				{prevChartDataSetRef.current === undefined && (
+					<Spinner
+						height="12vh"
+						tip="Loading..."
+						customStyle={{
+							justifyContent: 'flex-end',
+							marginRight: '5px',
+						}}
+					/>
+				)}
+				{!isEmpty(widget) && prevChartDataSetRef.current && (
```

Comment:
@csudani7 in the loading state, still there is a flash 

try to remove the if condition for loading state and add the check the rendering state only.

Reply:
@palashgdev - Updated my logic. Developed behavior as per grafana.